### DISCUSSION TITLE
[@types/pdfkit] TypeError: Cannot read properties of undefined (reading 'valueOf')

### DISCUSSION THREAD

**Discussion by Enough7** [DefinitelyTyped/DefinitelyTyped#71172] [Issues with a @types package] [answered]
Hi,
I think [`PDFKit.DocumentInfor.Title`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fd5ace7edf32dd34ce3fc86e6e2adb0d8e3a6a8d/types/pdfkit/index.d.ts#L655) is not correctly typed.
It throws a TypeError when I use
```Typescript
new PDFDocument(
        {
            size: "A4",
            info: {
                Title: undefined,
                Author: "Some Name"
            }
        }
    );
```
`PDFKit.DocumentInfor.Title`

It collides with this [check](https://github.com/foliojs/pdfkit/blob/4bad866564234efb44b1ab899d45657694644862/lib/security.js#L13-L19).

Error:
```
TypeError: Cannot read properties of undefined (reading 'valueOf')
    at Function.valueOf [as generateFileID] (/home/userp/Repositories/Privat/CVResumeJSON/Server/node_modules/pdfkit/lib/security.js:18:39)
    at new generateFileID (/home/userp/Repositories/Privat/CVResumeJSON/Server/node_modules/pdfkit/lib/document.js:118:28)
    at generatePDF (<Path of my code>)
```

## Suggestion
I guess [`PDFKit.DocumentInfor.Title`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fd5ace7edf32dd34ce3fc86e6e2adb0d8e3a6a8d/types/pdfkit/index.d.ts#L655) should be typed like this:
```Typescript
interface DocumentInfo {
    Producer?: string;
    Creator?: string;
    CreationDate?: Date;
    Title?: string;
    Author?: string;
    Subject?: string;
    Keywords?: string;
    ModDate?: Date;
}
```

[a comment by an automated account is not shown]

**Comment by BamButz** [comment, marked as answer]
I agree with your take.  If you're up for it, feel free to submit a pull request!
